src/ert/shared/models/base_run_model.py

```diff
@@ -430,38 +433,31 @@ def id_(self, value: str) -> None:
     async def _ensemble_listener(self, ensemble: Ensemble, iter_: int) -> None:
         """Redirect events emitted by the ensemble to this experiment."""
         while True:
-            event: CloudEvent = await ensemble.output_bus.get()
-            await self.dispatch(event, iter_)
-            if event["type"] in (
-                identifiers.EVTYPE_ENSEMBLE_FAILED,
-                identifiers.EVTYPE_ENSEMBLE_CANCELLED,
-                identifiers.EVTYPE_ENSEMBLE_STOPPED,
-            ):
-                break
-
-    async def _dispatch_ee(
-        self, ee_event_type: str, iteration: int, data: Optional[dict] = None
+            event: com_protocol.DispatcherMessage = await ensemble.output_bus.get()
+            await self.dispatch(event)
+            if event.WhichOneof("object") == "ensemble":
+                if event.ensemble.status in (
+                    com_protocol.ENSEMBLE_FAILED,
+                    com_protocol.ENSEMBLE_CANCELLED,
+                    com_protocol.ENSEMBLE_STOPPED,
+                ):
+                    break
+
+    @singledispatchmethod
+    async def dispatch(
+        self,
+        event: Union[CloudEvent, com_protocol.DispatcherMessage],
     ) -> None:
-        await self.dispatch(
-            CloudEvent(
-                {
-                    "type": ee_event_type,
-                    "source": f"/ert/experiment/{self.id_}",
-                    "id": str(uuid.uuid1()),
-                },
-                data,
-            ),
-            iteration,
-        )
+        raise NotImplementedError("Not implemented")
 
-    async def dispatch(self, event: CloudEvent, iter_: int) -> None:
+    @dispatch.register
+    async def _(self, event: CloudEvent) -> None:
+        event_logger.debug("dispatch cloudevent: %s (experiment: %s)", event, self.id_)
 
-        event_logger.debug(
-            "dispatch: %s (experiment: %s, iter: %d)", event, self.id_, iter_
-        )
-        if event["type"] == identifiers.EVTYPE_FM_STEP_SUCCESS:
-            real = int(get_real_id(event["source"]))
-            self._state_machine.add_successful_realization(iter_, real)
+    @dispatch.register
+    async def _(self, event: com_protocol.DispatcherMessage) -> None:
+        event_logger.debug(f"dispatch pbuf {event.WhichOneof('object')}")
```

Comment:
Should this debug be a little bit more detailed if we are to get anything useful out from it?
I have tried     `f"dispatch pbuf {event.WhichOneof('object')} {' '.join(str(event).split())}"` but it might be too much again.

Reply:
I've removed the logging